.eas/workflows/updates-e2e-enabled.yml

```diff
@@ -0,0 +1,151 @@
+        env:
+          GIT_AUTHOR_NAME: "Douglas Lowder"
+          GIT_AUTHOR_EMAIL: "douglowder@mac.com"
+        run: >
+          node ./packages/expo-updates/bin/check-for-changed-paths/index.js
+          ${{ github.event_name }}
+          main
+          .eas/workflows/updates-e2e-enabled.yml
+          'packages/@expo/cli/**'
+          'packages/@expo/config-plugins/**'
+          'packages/@expo/env/**'
+          'packages/@expo/metro-config/**'
+          'packages/@expo/prebuild-config/**'
+          'packages/expo/**'
+          'packages/expo-asset/**'
+          'packages/expo-font/**'
+          'packages/expo-json-utils/**'
+          'packages/expo-manifests/**'
+          'packages/expo-modules-autolinking/**'
+          'packages/expo-modules-core/**'
+          'packages/expo-splash-screen/**'
+          'packages/expo-structured-headers/**'
+          'packages/expo-updates-interface/**'
+          'packages/expo-updates/**'
+          'templates/expo-template-bare-minimum/**'
+          packages/expo-updates/e2e/setup/create-eas-project.ts
+          packages/expo-updates/e2e/setup/project.ts
+      - name: Print result
+        run: |
+          echo ${{ steps.changed_paths.outputs.result }}
+  ios:
+    runs_on: macos-large
+    needs: [check_paths]
+    if: ${{ needs.check_paths.outputs.result == 'true' }}
+    image: latest
+    steps:
+      - name: Install applesimutils
+        id: ios_simulator
+        env:
+          HOMEBREW_NO_AUTO_UPDATE: "1"
+        run: |
+          /opt/homebrew/bin/brew tap wix/brew
+          /opt/homebrew/bin/brew install applesimutils
+          xcrun simctl list
+      - uses: eas/checkout
+      - uses: eas/use_npm_token
+      - uses: eas/install_node_modules
+      - name: Set up Updates E2E enabled project
+        id: setup
+        env:
+          UPDATES_HOST: localhost
+          UPDATES_PORT: "4747"
+        run: |
+          yarn --silent ts-node --transpile-only ./packages/expo-updates/e2e/setup/create-eas-project.ts
+          ls -la ../updates-e2e
+      - name: Prepare E2E project
+        id: prepare
+        working_directory: ../updates-e2e
+        run: |
+          yarn generate-test-update-bundles
+          npx pod-install
+      - name: Build E2E project (debug and release)
+        id: build
+        working_directory: ../updates-e2e
+        run: |
+          npx detox build -c ios.debug
+          npx detox build -c ios.release
+      - name: Test E2E project (debug and release)
+        id: test
+        env:
+          UPDATES_HOST: localhost
+          UPDATES_PORT: "4747"
+        working_directory: ../updates-e2e
+        run: |
+          npx detox test -c ios.debug          
+          npx detox test -c ios.release
+  android:
+    runs_on: linux-large-nested-virtualization
+    needs: [check_paths]
+    if: ${{ needs.check_paths.outputs.result == 'true' }}
+    steps:
+      - name: Set up JDK 17 and Android dependencies
```

Comment:
I would hope you could get rid of it if you used one of the JDK17 images? (And manual `JAVA_HOME`s later in the workflow run.)

https://github.com/expo/universe/blob/2aa736dd9512b1fd58727fe6bbe830275ef6b4ba/server/www/src/utils/OrchestratorUtils.ts#L96-L115